### PR TITLE
Add --selector support to osadm registry

### LINKED ISSUE #2451: osadm registry doesn't support --selector flag
Like the router, users may wish to specify where their registry runs by providing a nodeSelector via `--selector` flag when creating the registry.

### PR DESCRIPTION
**Pull request by ncdc**
Fixes #2451

### CODE REVIEW

**Comment by ncdc** [comment]
@brenton @ramr PTAL

**Comment by brenton** [comment]
Looks like it would do the job.  I'm guessing we'd have to refactor our osadm commands a bit for the router and registry to be able to share the node selector logic.

File: pkg/cmd/experimental/registry/registry.go [hunk 1]
```diff
@@ -140,6 +143,18 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		label = valid
 	}
 
+	nodeSelector := map[string]string{}
+	if len(cfg.Selector) > 0 {
+		valid, remove, err := app.LabelsFromSpec(strings.Split(cfg.Selector, ","))
+		if err != nil {
+			glog.Fatal(err)
```
Comment (ramr) [inline pkg/cmd/experimental/registry/registry.go]: ```return err``` would be more consistent here - similar to the code block above.

**Comment by ramr** [comment]
LGTM - minor nitpick about the error return and check on the test failures.

File: pkg/cmd/experimental/registry/registry.go (hunk 1 above)
Comment (ncdc) [inline pkg/cmd/experimental/registry/registry.go]: I copied this from the router. Looks like there is a discrepancy between the 2. Router calls `glog.Fatal` whereas Registry returns err.

Comment (liggitt) [inline pkg/cmd/experimental/registry/registry.go]: should always bubble errors to a checkErr at the top level of the command

Comment (ncdc) [inline pkg/cmd/experimental/registry/registry.go]: I’ll make the change here and create an issue to update the router and any other commands.

**Comment by ncdc** [comment]
Test failure from Travis:
```
error: timed out waiting for the condition
!!! Error in hack/test-cmd.sh:572
  'osc delete all --all' exited with status 1
```

Worked for me on my laptop.

File: pkg/cmd/experimental/registry/registry.go (hunk 1 above)
Comment (ramr) [inline pkg/cmd/experimental/registry/registry.go]: Yeah, the router code doesn't bubble up the errors - that needs to be fixed sometime. 

As @liggitt  mentioned, the preferred way would be to bubble up the errors and have the top level code (```cobra.Command :: Run``` handle it)  ala: 
```  
 Run: func(cmd *cobra.Command, args []string) {  
        err := processCommand(f, options, cmd, args, out)  
        cmdutil.CheckErr(err)  
},  
```

**Comment by ncdc** [comment]
error return fixed

**Comment by ncdc** [comment]
[merge]

**Pull request merged**